fog/report/types/src/lib.rs

```diff
@@ -9,22 +9,47 @@
 
 extern crate alloc;
 
+use ::prost::Message;
 use alloc::{collections::BTreeMap, string::String, vec::Vec};
-use mc_attest_core::VerificationReport;
+use mc_attest_verifier_types::{prost, VerificationReport};
 use mc_crypto_digestible::Digestible;
-use prost::Message;
 use serde::{Deserialize, Serialize};
 
+/// The attestation evidence variants for a report.
+#[derive(Clone, ::prost::Oneof, Deserialize, Eq, PartialEq, Serialize, Digestible)]
+#[digestible(transparent)]
+pub enum AttestationEvidence {
+    /// The attestation evidence is a [VerificationReport].
+    #[prost(message, tag = 2)]
+    VerificationReport(VerificationReport),
+    /// DCAP evidence
+    #[prost(message, tag = 4)]
+    DcapEvidence(prost::DcapEvidence),
+}
+
+impl From<VerificationReport> for AttestationEvidence {
+    fn from(report: VerificationReport) -> Self {
+        Self::VerificationReport(report)
+    }
+}
+
+impl From<prost::DcapEvidence> for AttestationEvidence {
+    fn from(evidence: prost::DcapEvidence) -> Self {
+        Self::DcapEvidence(evidence)
+    }
+}
+
 /// A fog report from the report server
 #[derive(Clone, Digestible, Eq, PartialEq, Serialize, Deserialize, Message)]
 pub struct Report {
     /// The fog_report_id of the report
     #[prost(string, tag = "1")]
     #[digestible(never_omit)]
     pub fog_report_id: String,
-    /// The bytes of the verification report
-    #[prost(message, required, tag = "2")]
-    pub report: VerificationReport,
+    /// Attestation evidence for the enclave.
+    #[prost(oneof = "AttestationEvidence", tags = "2, 4")]
+    #[digestible(name = "report")]
+    pub attestation_evidence: Option<AttestationEvidence>,
```

Comment:
I did not add a backwards compatible test here, and I'm wondering if we need the digestible naming.
It looks like `Digestible` is only used for signing the `Report`. 

Maybe we need the backwards compatibility for clients to be able to check the signature on both versions durging the EPID to DCAP transition

I am kind of curious on the history here as the protobufs expose the signature and at first blush I would think the signature in the protobuf was over a sha of the report bytes in the protobuf.

Reply:
NVM added a backwards test to ensure it wasn't this causing test failures